solr/modules/s3-repository/src/java/org/apache/solr/s3/S3BackupRepositoryConfig.java

```diff
@@ -59,7 +58,7 @@ public S3StorageClient buildClient() {
   }
 
   private static String getStringConfig(NamedList<?> config, String property) {
-    String envProp = System.getenv().get(toEnvVar(property));
+    String envProp = EnvUtils.getProp(property);
```

Comment:
No need for `toEnvVar(prop)` here (which does `solr.foo.bar -> SOLR_FOO_BAR`) since we can rely on the fact that `EnvUtils.getProp` will obtain `solr.foo.bar` even when set as env `SOLR_FOO_BAR`.

Reply:
Nice!